patches/minecraft/net/minecraft/inventory/container/AbstractFurnaceContainer.java.patch

```diff
@@ -0,0 +1,12 @@
+--- a/net/minecraft/inventory/container/AbstractFurnaceContainer.java
++++ b/net/minecraft/inventory/container/AbstractFurnaceContainer.java
+@@ -171,4 +171,9 @@
+    public boolean func_217061_l() {
+       return this.field_217064_e.func_221476_a(0) > 0;
+    }
++
++   @Override
++   public IRecipeType<?> getRecipeType() {
```

Comment:
You should move this over to the class where you added this method and use `instanceof` checks. It's not the best OOP design, but it saves up on patch count. Same for `PlayerContainer`, `WorkbenchContainer`

Reply:
implemented default body for getRecipeType